### PR TITLE
Add support for guarded transactions

### PR DESCRIPTION
**Pull request by andreibancioiu**

### CODE REVIEW

File: src/MultiversX/NetworkConfig.cpp [hunk 1]
```diff
@@ -38,6 +38,14 @@ void NetworkConfig::setMinGasLimit(uint32_t value) {
     this->minGasLimit = value;
 }
 
+uint32_t NetworkConfig::getExtraGasLimitForGuardedTransaction() const {
+    return this->minGasLimit;
+}
+
+void NetworkConfig::setExtraGasLimitForGuardedTransaction(uint32_t value) {
+    this->minGasLimit = value;
```
Comment (AdoAdoAdo) [inline src/MultiversX/NetworkConfig.cpp]: I don't think this is correct as this is overwriting the minGasLimit

File: src/MultiversX/GasEstimator.h [hunk 2]
```diff
@@ -6,19 +6,20 @@
...
 class GasEstimator {
     NetworkConfig networkConfig;
+
 public:
     GasEstimator(const NetworkConfig& networkConfig);
 
-    uint64_t forEGLDTransfer(size_t dataLength) const;
-    uint64_t forESDTTransfer(size_t dataLength) const;
-    uint64_t forESDTNFTTransfer(size_t dataLength) const;
+    uint64_t forEGLDTransfer(size_t dataLength, bool isGuarded) const;
```
Comment (AdoAdoAdo) [inline src/MultiversX/GasEstimator.h]: maybe use different method instead of bool argument?

File: src/MultiversX/NetworkConfig.cpp (hunk 1 above)
Comment (andreibancioiu) [reply src/MultiversX/NetworkConfig.cpp]: Indeed, good catch :)

File: src/MultiversX/GasEstimator.h (hunk 2 above)
Comment (andreibancioiu) [reply src/MultiversX/GasEstimator.h]: Yes, will add separate methods :+1:

**Review by ccorcoveanu — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Pull request closed without merging**